generate.sh

```diff
@@ -68,26 +74,30 @@ parse_rules() {
                         continue
                     fi
                 fi
+                prolog="\$(V_DGM)$cdir"
                 core_subst=$core
                ;;
             link)
                 [[ -z "$link_name" ]] && echo "link_name has to be specified in link rule" && exit 1
                 [[ -z "$link_target" ]] && echo "link_target has to be specified in link rule" && exit 1
                 dest="$link_name"
                 core_subst=$(echo $core | sed -e "s~__link_target__~"${link_target}"~g")
+                prolog="\$(V_LN)$cdir"
                 ;;
         esac
-        t=$(echo ${DESTDIR}/${version}/${dest} | sed 's/ /\\ /g')
-        src=$(echo $src | sed 's/ /\\ /g' )
-        targets+="	$t \\
-"
-        cat >> auto_targets.mk << EOF
-$t: $src manifest.sh
-	@echo ${message} ; \\
-	mkdir -p "\$\$(dirname \$@)" || exit 1 ; \\
-	${core_subst}
-    ${mode:+"chmod ${mode} "\$@""}
 
+        case $version$dest$src in
+            *' '*) die "space not allowed in version, dest, src" ;;
+        esac
+
+        target=$version/$dest
+        targets+="\\$nl	$target"
+
+        cat >> auto_targets.mk << EOF
+$nl$target: $src manifest.sh
```

Comment:
This should depend on `$MANIFEST_FILE ` instead of `manifest.sh`

Reply:
Very valid point, thanks!